### PR TITLE
Migrate ESTestCase.copy* and AbstractBWC test cases to TransportVersion

### PR DESCRIPTION
**Pull request by thecoop**
Migrate `ESTestCase.copy*` methods to TransportVersion, propagating through `AbstractBWC*SerializationTestCase`.

Majority of the changes are `s/Version/TransportVersion`

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModelLocation.java [hunk 1]
```diff
@@ -7,16 +7,10 @@
...
-import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.NamedWriteable;
-import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
 import org.elasticsearch.xpack.core.ml.utils.NamedXContentObject;
 
 public interface TrainedModelLocation extends NamedXContentObject, NamedWriteable {
 
     String getResourceName();
-
-    default Version getMinimalCompatibilityVersion() {
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModelLocation.java:19]: This method is not actually used anywhere

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ public class TrainedModelConfig implements ToXContentObject, Writeable {
     public static final ParseField INFERENCE_CONFIG = new ParseField("inference_config");
     public static final ParseField LOCATION = new ParseField("location");
 
-    public static final Version VERSION_3RD_PARTY_CONFIG_ADDED = Version.V_8_0_0;
+    public static final TransportVersion VERSION_3RD_PARTY_CONFIG_ADDED = TransportVersion.V_8_0_0;
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java:100]: This and `TrainedModelLocation` is the only non-test change in this PR

File: x-pack/plugin/core/src/test/java/org/elasticsearch/protocol/xpack/XPackInfoRequestTests.java [hunk 3]
```diff
@@ -21,21 +21,20 @@
...
-        // 7.8.1 (inclusive) and 8.0.0 (exclusive), because the "version before 8.0.0" returns 7.8.0 (the most recent released version).
-        // To work around this we accept that 8.0.0 is included in the range, and then filter it out using other-than
-        final Version version = randomValueOtherThan(
-            Version.V_8_0_0,
-            () -> VersionUtils.randomVersionBetween(random(), Version.V_7_8_1, Version.V_8_0_0)
+        assertSerialization(
+            TransportVersionUtils.randomVersionBetween(
+                random(),
+                TransportVersion.V_7_8_1,
+                TransportVersionUtils.getPreviousVersion(TransportVersion.V_8_0_0)
```
Comment (thecoop) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/protocol/xpack/XPackInfoRequestTests.java:32]: Rewritten logic to use `getPreviousVersion`

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModelLocation.java (hunk 1 above)
Comment (thecoop) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModelLocation.java:19]: @davidkyle is it ok this is not actually used?

Comment (davidkyle) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TrainedModelLocation.java:19]: Yes please delete it. 

The streaming compatibility check is performed elsewhere, it's not needed here.
Thanks
https://github.com/elastic/elasticsearch/blob/6d009a38c79c2bdea764a4f567671e3e3827700a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java#L250

**Review by pgomulka — APPROVED** [review]
LGTM

**Pull request merged**